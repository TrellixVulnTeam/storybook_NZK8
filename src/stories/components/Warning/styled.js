// @flow

import styled from 'styled-components';
import AlertIcon from './img/icon-alert.svg';

export const Alert = styled(AlertIcon)`
  height: 20px;
  width: 20px;
  margin-right: 10px;
`;

export const WarningMessage = styled.div`
  padding: 10px 18px;
  background-color: rgba(224, 170, 8, 0.05);
  border-radius: 4px;
  font-size: 12px;
  display: flex;
  align-items: center;
  margin-bottom: 2rem;
`;
