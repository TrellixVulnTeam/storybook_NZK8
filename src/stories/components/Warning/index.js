// @flow

import React, { PureComponent } from 'react';

import { WarningMessage, Alert } from './styled';

type Props = {
  message?: string,
};
class Warning extends PureComponent<Props> {
  /**
   * Render a modal with a form for adding a new user
   * @return {React$Node}
   */
  render() {
    const { message } = this.props;
    return (
      <WarningMessage>
        <Alert /> {message || 'You have exceeded your plan limit. Upgrade your plan today!'}
      </WarningMessage>
    );
  }
}

export default Warning;
