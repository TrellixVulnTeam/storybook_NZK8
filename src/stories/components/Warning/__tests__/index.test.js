import React from 'react';
import { mount } from 'enzyme';
import Warning from 'components/Warning';

describe('<Warning />', () => {
  it('should render correctly with default warning text', () => {
    const sut = mount(<Warning />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with custom warning text', () => {
    const sut = mount(<Warning message="Test message" />);

    expect(sut).toMatchSnapshot();
  });
});
