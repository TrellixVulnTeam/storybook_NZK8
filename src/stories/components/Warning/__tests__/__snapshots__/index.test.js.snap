// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Warning /> should render correctly with custom warning text 1`] = `
<Warning
  message="Test message"
>
  <styled__WarningMessage>
    <div
      className="styled__WarningMessage-sc-1dzobon-1 hdedCK"
    >
      <styled__Alert>
        <test-file-stub
          className="styled__Alert-sc-1dzobon-0 fsLAJZ"
        />
      </styled__Alert>
       
      Test message
    </div>
  </styled__WarningMessage>
</Warning>
`;

exports[`<Warning /> should render correctly with default warning text 1`] = `
<Warning>
  <styled__WarningMessage>
    <div
      className="styled__WarningMessage-sc-1dzobon-1 hdedCK"
    >
      <styled__Alert>
        <test-file-stub
          className="styled__Alert-sc-1dzobon-0 fsLAJZ"
        />
      </styled__Alert>
       
      You have exceeded your plan limit. Upgrade your plan today!
    </div>
  </styled__WarningMessage>
</Warning>
`;
